Occurrence over night

ch000 opened the pull request:
I made a method which sets the end_time-date one day later than the start_time-date, if @params[:end_time] <= @params[:start_time].

So it is possible to create a schedule over night (e.g. from 8pm to 2 am).

dgilperez committed:
upgrade icecube to 0.11.2

Commit:
added end_time_date

Commit:
added end_time_date

Commit:
end_time is tomorrow if its equal to start_time

dgilperez committed:
Gemfile.lock

dgilperez committed:
add test to catch infinite loop

dgilperez committed:
fixing issue #18. Calling validations_for(:month_of_year), even if no validation applies, will alter Schedule::Rule, initializing @validations[:month_of_year] with an empty array. Both IceCube and schedule_attributes will benefit for some refactoring of these areas, but this fixes this for now

dgilperez committed:
bump ice_cube dependency to >= 0.12 in gemspec

dgilperez committed:
Rails 4.1 compatiblity: Date.current.to_time_in_current_zone is now Date.in_time_zone

ch000 committed:
Merge pull request #1 from dgilperez/master

Upgrade IceCube & Rails4 compatibility

avit commented:
My own use-case does not include overnight events so I'm not sure if these semantics would be confusing, but this may be worth rebasing and restricting to only the relevant changes here.

See also: #4

ch000 committed:
fix: couldn't save with endless schedule (no start&end-date) which occur over night

i tried to parse the date to set the end_time-date. But the date doesn't have to exist for endless schedules. Using `parse_date_time` it creates a current date if no date is given.

ch000 committed:
Update input.rb

ch000 committed:
Update active_record.rb

ch000 committed:
Update ice_cube.rb
